core/src/main/java/org/elasticsearch/plugins/DiscoveryPlugin.java

```diff
@@ -52,4 +56,6 @@
     default NetworkService.CustomNameResolver getCustomNameResolver(Settings settings) {
         return null;
     }
+
+    default Set<Class<? extends ZenPing>> getZenPings(Settings settings) { return Collections.emptySet(); }
```

Comment:
And if this is only there for testing, then it should be done in a way that only tests can override (eg as an override in MockNode).